Linux failure might be the expected failure until after https://github.com/apple/swift-corelibs-foundation/pull/1441#issuecomment-366554616 is merged
```
Test Case 'TestNSString.test_PrefixSuffix' started at 2018-02-18 22:55:45.216
TestFoundation/TestNSString.swift:1412: error: TestNSString.test_PrefixSuffix : XCTAssertTrue failed - Unexpected success: 1241
```

--- a/stdlib/public/core/StringComparison.swift
+++ b/stdlib/public/core/StringComparison.swift
@@ -24,13 +24,13 @@ func _compareUnicode(
   let left = _StringGuts(rawBits: lhs)
   let right = _StringGuts(rawBits: rhs)
 
-   if _slowPath(!left._isContiguous || !right._isContiguous) {
-     if !left._isContiguous {
-       return left._asOpaque()._compareOpaque(right).rawValue
-     } else {
-       return right._asOpaque()._compareOpaque(left).flipped.rawValue
-     }
-   }
+  if _slowPath(!left._isContiguous || !right._isContiguous) {
+    if !left._isContiguous {
+      return left._asOpaque()._compareOpaque(right).rawValue
+    } else {
+      return right._asOpaque()._compareOpaque(left).flipped.rawValue
+    }
+  }
 
   return left._compareContiguous(right)
 }

--- a/validation-test/stdlib/String.swift
+++ b/validation-test/stdlib/String.swift
@@ -1985,16 +1985,14 @@ struct ComparisonTestCase {
   }
 }
 
-let simpleComparisonTestCases = [
+let comparisonTestCases = [
   ComparisonTestCase(["a", "a"], .equal),
   ComparisonTestCase(["abcdefg", "abcdefg"], .equal),
   ComparisonTestCase(["", "Z", "a", "b", "c", "\u{00c5}", "á"], .less),
 
   ComparisonTestCase(["ábcdefg", "ábcdefgh", "ábcdefghi"], .less),
   ComparisonTestCase(["abcdefg", "abcdefgh", "abcdefghi"], .less),
-]
 
-let complexComparisonTestCases = [
   ComparisonTestCase(["á", "\u{0061}\u{0301}"], .equal),
   ComparisonTestCase(["à", "\u{0061}\u{0301}", "â", "\u{e3}", "a\u{0308}"], .less),
   
@@ -2066,8 +2064,6 @@ let complexComparisonTestCases = [
   ComparisonTestCase(["ì̡̢̧̨̝̞̟̠̣̤̥̦̩̪̫̬̭̮̯̰̹̺̻̼͇͈͉͍͎́̂̃̄̉̊̋̌̍̎̏̐̑̒̓̽̾̿̀́͂̓̈́͆͊͋͌ͅ͏͓͔͕͖͙͐͑͒͗ͬͭͮ͘", "ì̡̢̧̨̝̞̟̠̣̤̥̦̩̪̫̬̭̮̯̰̹̺̻̼͇͈͉͍͎́̂̃̄̉̊̋̌̍̎̏̐̑̒̓̽̾̿̀́͂̓̈́͆͊͋͌ͅ͏͓͔͕͖͙͐͑͒͗ͬͭͮ͘"], .equal)
 ]
 
-let comparisonTestCases = simpleComparisonTestCases + complexComparisonTestCases
-
 for test in comparisonTestCases {
   StringTests.test("Comparison.\(test.strings)") {
     test.test()
@@ -2078,9 +2074,7 @@ for test in comparisonTestCases {
     .code {
     test.testOpaqueStrings()
   }
-}
 
-for test in simpleComparisonTestCases {
   StringTests.test("Comparison.OpaqueSubstring.\(test.strings)")
     .skip(.linuxAny(reason: "NSSlowString requires ObjC interop"))
     .code {

